app/models/api.rb

```diff
@@ -0,0 +1,5 @@
+class Api < ActiveRecord::Base
+  validates :name, presence: true, uniqueness: true
+  validates :url, presence: true, uniqueness: true
```

Comment:
It might be interesting to do a dependent uniqueness validation here. Presumably the name and url would need to be the same in combination for a duplicate to exist, correct?